test/wasm2js/i64-add-sub.2asm.js.opt

```diff
@@ -21,7 +21,7 @@ function asmFunc(env) {
  }
  
  function legalstub$2($0, $1, $2, $3, $4, $5) {
-  return ($4 | 0) == ($0 - $2 | 0) & ($5 | 0) == ($1 - (($0 >>> 0 < $2 >>> 0) + $3 | 0) | 0);
+  return ($0 - $2 | 0) == ($4 | 0) & ($1 - (($0 >>> 0 < $2 >>> 0) + $3 | 0) | 0) == ($5 | 0);
  }
```

Comment:
do you know why this changes?

Reply:
I also added extra optimization pass after optimize-for-js:
https://github.com/WebAssembly/binaryen/pull/4055/files#diff-503ab0ba884eaf72c574346fa45a6875959b6b1b3c90787a7d3ce4d7f6b464f6R348

this necessary due to this rule skip some cases which already can handled by "optimize-instructions" pass